sklearn/preprocessing/tests/test_data.py

```diff
@@ -818,89 +812,75 @@ def test_min_max_scaler_1d():
 
 
 @pytest.mark.parametrize("sample_weight", [True, None])
-def test_scaler_without_centering(sample_weight):
+@pytest.mark.parametrize("sparse_container", [*CSC_CONTAINERS, *CSR_CONTAINERS])
+def test_scaler_without_centering(sample_weight, sparse_container):
     rng = np.random.RandomState(42)
     X = rng.randn(4, 5)
     X[:, 0] = 0.0  # first feature is always of zero
-    X_csr = sparse.csr_matrix(X)
-    X_csc = sparse.csc_matrix(X)
+    X_sparse = sparse_container(X)
 
     if sample_weight:
         sample_weight = rng.rand(X.shape[0])
 
     with pytest.raises(ValueError):
-        StandardScaler().fit(X_csr)
-    with pytest.raises(ValueError):
-        StandardScaler().fit(X_csc)
+        StandardScaler().fit(X_sparse)
 
-    null_transform = StandardScaler(with_mean=False, with_std=False, copy=True)
-    X_null = null_transform.fit_transform(X_csr)
-    assert_array_equal(X_null.data, X_csr.data)
-    X_orig = null_transform.inverse_transform(X_null)
-    assert_array_equal(X_orig.data, X_csr.data)
+    if sparse_container in CSR_CONTAINERS:
```

Comment:
Could you move this `if` at the end of the test to have the common part altogether.

Reply:
If it seems weird, we could isolate the test since this is really specific.

Reply:
I've tried to make it a separate test, but not sure how to name it. Please let me know if there is better naming suggestion.